stock_location_children/models/stock_location.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, fields, models
+
+
+class StockLocation(models.Model):
+
+    _inherit = 'stock.location'
+
+    children_ids = fields.Many2many(
+        'stock.location',
+        'stock_location_children_ids',
+        'parent_id',
+        'children_id',
+        compute='_compute_children_ids',
+        store=True,
+        help='All the children (multi-level) stock location of this location',
+    )
+
+    @api.depends('child_ids', 'child_ids.children_ids')
+    def _compute_children_ids(self):
+        for loc in self:
```

Comment:
can't we use `parent_path` to make this more performant?
Can you check its usage in stock module?